app/main.py

```diff
@@ -1,3 +1,80 @@
+from typing import Union
+
+
 class Distance:
-    # Write your code here
-    pass
+
+    def __init__(self, km: (int, float)) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        if self.km == 1:
+            return f"Distance: {self.km} kilometer."
+        else:
+            return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: "Distance") -> "Distance":
+        if isinstance(other, Distance):
+            return Distance(km=self.km + other.km)
+        elif isinstance(other, (float, int)):
+            return Distance(km=self.km + other)
```

Comment:
The `__add__` method should have a type hint that reflects it can handle both `Distance` instances and numeric types, similar to the `__iadd__` method.